include/deal.II/base/aligned_vector.h

```diff
@@ -1199,11 +1199,22 @@ template <class T>
 inline AlignedVector<T> &
 AlignedVector<T>::operator=(const AlignedVector<T> &vec)
 {
+  const size_type new_size = vec.used_elements_end - vec.elements.get();
+
+  // First throw away everything and re-allocate memory but leave that
+  // memory uninitialized for now:
   resize(0);
-  resize_fast(vec.used_elements_end - vec.elements.get());
+  reserve(new_size);
```

Comment:
The bug is here: When we call `resize_fast()`, we allocate memory and default construct objects in that memory. But in the line below (`AlignedVectorCopyConstruct`) we copy-construct *other* objects in that same memory location, forgetting to call the destructor on what we had done before. 

The solution is to not initialize the memory we allocate -- that's what `reserve()` does in contrast to `resize*()`.

Reply:
For reference, on second look, I see that in essence the same code I just wrote here is also what's in the copy constructor just a few lines up.